src/Runner.Worker/JobRunner.cs

```diff
@@ -145,6 +145,11 @@ public async Task<TaskResult> RunAsync(Pipelines.AgentJobRequestMessage message,
                 Trace.Verbose($"Job steps: '{string.Join(", ", jobSteps.Select(x => x.DisplayName))}'");
                 HostContext.WritePerfCounter($"WorkerJobInitialized_{message.RequestId.ToString()}");
 
+                // We want to let the job is marked as in-progress before running any customer's steps as much as we can.
+                // Timeline record update background process runs every 500ms, so delay 1000ms is enough for most of the cases
+                Trace.Info($"Waiting for job to be marked as started.");
+                await Task.WhenAny(_jobServerQueue.JobRecordUpdated.Task, Task.Delay(1000));
```

Comment:
Is that going to delay all job invocations?

Reply:
🙇  I changed it to only delay jobs that uses OIDC.